include/boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp

```diff
@@ -91,7 +91,7 @@ struct range_iterator_end<Range, open>
 
 
 template <typename Range, typename Value, typename Reference = Value>
-class range_segment_iterator
+struct range_segment_iterator
```

Comment:
NOTE: apprently this gives errors for clang because elsewhere you have declared this as a friend.

````
"clang++" -x c++ -Wno-unneeded-internal-declaration -std=c++11 -O0 -g -O0 -fno-inline -Wall -g -DBOOST_ALL_NO_LIB=1 -I".." -I"../boost/geometry/extensions/contrib/ttmath" -I"../libs/geometry/test" -c -o "/Volumes/Data/kbelco/boost_testing/results/boost/bin.v2/libs/geometry/test/iterators/segment_iterator.test/clang-darwin-4.2.1/debug/segment_iterator.o" "../libs/geometry/test/iterators/segment_iterator.cpp"

In file included from ../libs/geometry/test/iterators/segment_iterator.cpp:46:
In file included from ../boost/geometry/iterators/segment_iterator.hpp:22:
In file included from ../boost/geometry/iterators/detail/segment_iterator/iterator_type.hpp:27:
../boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp:157:12: warning: class template 'range_segment_iterator' was previously declared as a struct template [-Wmismatched-tags]
    friend class range_segment_iterator;
           ^
../boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp:94:8: note: previous use is here
struct range_segment_iterator
       ^
../boost/geometry/iterators/detail/segment_iterator/range_segment_iterator.hpp:157:12: note: did you mean struct here?
    friend class range_segment_iterator;
           ^~~~~
           struct
1 warning generated.
            
```

Please fix before we can merge this to master...